test/python/pulse/test_transforms.py

```diff
@@ -401,12 +407,286 @@ def test_multiple_schedules(self):
             schedule += Play(SamplePulse([0.0, 0.2]), drive_channel)
             schedules.append(schedule)
 
-        compressed_schedule = compress_pulses(schedules)
+        compressed_schedule = transforms.compress_pulses(schedules)
         original_pulse_ids = get_pulse_ids(schedules)
         compressed_pulse_ids = get_pulse_ids(compressed_schedule)
         self.assertEqual(len(original_pulse_ids), 6)
         self.assertEqual(len(compressed_pulse_ids), 2)
 
 
+class TestAlignSequential(QiskitTestCase):
+    """Test sequential alignment transform."""
+
+    def test_align_sequential(self):
+        """Test sequential alignment without a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule = schedule.insert(4, instructions.Delay(5, d1), mutate=True)
+        schedule = schedule.insert(12, instructions.Delay(7, d0), mutate=True)
+
+        schedule = transforms.remove_directives(transforms.align_sequential(schedule))
+
+        reference = pulse.Schedule()
+        # d0
+        reference.insert(0, instructions.Delay(3, d0), mutate=True)
+        reference.insert(8, instructions.Delay(7, d0), mutate=True)
+        # d1
+        reference.insert(3, instructions.Delay(5, d1), mutate=True)
+
+        self.assertEqual(schedule, reference)
+
+    def test_align_sequential_with_barrier(self):
+        """Test sequential alignment with a barrier."""
```

Comment:
This looks similar to the last test. Even if the barrier is removed, the outcome would actually be the same, so are you sure you're testing anything here? Removing line 448 should cause this test to fail.

Also, I'm curious again why you have to `remove_directives` here?

Reply:
Fixed in 82300005a.